### DISCUSSION TITLE
(This already exists) A timer that switches the device off/on after n minutes

### DISCUSSION THREAD

**Discussion by kn100** [arendst/Tasmota#10559] [Ideas]
Hi all,

Absolutely love Tasmota, and am running it everywhere I can be. Thank you!

This is a feature request - that I'd be happy to look into implementing myself if given some pointers on implementation beforehand!

I'm imagining a simple feature where you could set a max time to be in a particular state, before it toggles. So, I could configure it so, 
* This plug can be on for a maximum of 30 minutes before it toggles off OR
* This plug can be off for a maximum of 60 minutes, before it toggles back on

My use case is this, I have an electric space heater that I control using a smart plug running Tasmota. Currently, I have a Raspberry Pi running Home Assistant which has an automation which times how long the plug has been on for, and switches the plug off after 30 minutes for safety. What if the Pi dies though? It would be really nice if even the network dies the plug would enforce this constraint.

Does this sound useful? Does it already exist?

[EDIT]
Yes it does already exist! I just didn't come across it in the docs. Search https://tasmota.github.io/docs/Commands/ for PulseTime

**Comment by ascillato** [comment]
Hi, already available. Please search in docs for command pulsetime and also check for rules. Thanks.

**Comment by barbudor** [comment]
Hi
You can have a look at `PulseTime` which can be used for one or the other (timer for on or timer for off) but not both.

Or you can use Rules triggered by power on/off orders coming from your home automation that trigger a RuleTimer for max duration of that state and cancel the RuleTimer for the previous state.

**Comment by meingraham** [comment]
Turning the device off after 30 minutes can be accomplished by setting `PulseTime<x>`. Turning it back on 60 minutes later can be accomplished using Rules and setting a trigger on `Power<x>#State` to detect power off and set a 60 minute rule timer. When the timer expires, it triggers a `Rules#Timer` rule to turn the device back on.

**Comment by kn100** [comment]
Thanks all, this is great news! I will look into it later today.

**Comment by MaxVonEvil** [comment]
Hi all old thread and all , I noticed the documentation for Pulsetime states: "Note if you have more than 8 relays: Defined PulseTime for relays <1-8> will also be active for correspondent Relay <9-16>."

I happen to have 11 relays under the control of an custom ESP32 based pool controller setup. Is there a good way to accomplish the same functionality for a relay #1 without affecting relay #9 ? I presume a rule can do the trick, but not entirely sure?

**Reply by sfromis** [reply]
The text about 1-8 is outdated. If you type the command `PulseTime` on ESP32, you should see that you have up to 32.

**Reply by MaxVonEvil** [reply]
Oh perfect, thanks for clarifying that!
